### ISSUE TITLE
Develop branch: error configuring as subproject

### ISSUE THREAD

**Issue by dmdunla** [kokkos/kokkos#3140]
When configuring Kokkos as a subproject, there is an error due to the fact that no project has been set. The error started with commit e9496424c17c3f15cf35bfb1cb285c598d67dbc9, as the removal of the "${}" around PROJECT_NAME means that the code that sets the project inside the conditional will never execute. Without the project being set, "${Kokkos_BINARY_DIR}", in turn will not be set and thus there is an error in ```cmake/kokkos_install.cmake``` (line 10) as shown below.

```
$ cmake .. -DKokkos_ENABLE_SERIAL:BOOL=ON
-- The CXX compiler identification is GNU 7.4.0
-- The C compiler identification is GNU 7.4.0
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Found PythonInterp: /usr/bin/python (found version "2.7.12") 
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Setting default Kokkos CXX standard to 11
-- Setting policy CMP0074 to use <Package>_ROOT variables
-- The project name is: Sparten
-- Using -std=c++11 for C++11 standard as feature
-- Execution Spaces:
--     Device Parallel: NONE
--     Host Parallel: NONE
--       Host Serial: SERIAL
-- 
-- Architectures:
-- Found TPLLIBDL: /usr/lib/x86_64-linux-gnu/libdl.so  
CMake Error: Could not open file for write in copy operation /KokkosConfig.cmake.tmp
CMake Error: : System Error: Permission denied
CMake Error at /usr/share/cmake-3.15/Modules/CMakePackageConfigHelpers.cmake:330 (configure_file):
  configure_file Problem configuring file
Call Stack (most recent call first):
  tpl/kokkos/cmake/kokkos_install.cmake:10 (CONFIGURE_PACKAGE_CONFIG_FILE)
  tpl/kokkos/CMakeLists.txt:262 (INCLUDE)
CMake Error: Could not open file for write in copy operation /KokkosConfigCommon.cmake.tmp
CMake Error: : System Error: Permission denied
CMake Error at /usr/share/cmake-3.15/Modules/CMakePackageConfigHelpers.cmake:330 (configure_file):
  configure_file Problem configuring file
Call Stack (most recent call first):
  tpl/kokkos/cmake/kokkos_install.cmake:15 (CONFIGURE_PACKAGE_CONFIG_FILE)
  tpl/kokkos/CMakeLists.txt:262 (INCLUDE)
CMake Error: Could not open file for write in copy operation /KokkosConfigVersion.cmake.tmp
CMake Error: : System Error: Permission denied
CMake Error at /usr/share/cmake-3.15/Modules/WriteBasicConfigVersionFile.cmake:49 (configure_file):
  configure_file Problem configuring file
Call Stack (most recent call first):
  /usr/share/cmake-3.15/Modules/CMakePackageConfigHelpers.cmake:225 (write_basic_config_version_file)
  tpl/kokkos/cmake/kokkos_install.cmake:20 (WRITE_BASIC_PACKAGE_VERSION_FILE)
  tpl/kokkos/CMakeLists.txt:262 (INCLUDE)
HWLOC_LIBRARIES = HWLOC_LIBRARIES-NOTFOUND
-- Found Doxygen: /usr/bin/doxygen (found version "1.8.11") found components:  doxygen missing components:  dot
CMAKE_VERSION = '3.15.2'
CMAKE_CXX_COMPILER = '/usr/bin/c++'
CMAKE_CXX_COMPILER_ID = 'GNU'
CMAKE_CXX_COMPILER_VERSION = '7.4.0'
CMAKE_CXX_STANDARD = ''
CMAKE_CXX_EXTENSIONS = 'OFF'
CMAKE_CXX_FLAGS = ''
CMAKE_CPP_FLAGS = ''
CMAKE_C_COMPILER = '/usr/bin/cc'
CMAKE_C_COMPILER_ID = 'GNU'
CMAKE_C_COMPILER_VERSION = '7.4.0'
CMAKE_C_FLAGS = ''
CMAKE_EXE_LINKER_FLAGS = ''
PROJECT_SOURCE_DIR = '/home/gitlab-runner/builds/7ab87121/0/questa-tensor/software/sparten'
PROJECT_BINARY_DIR = '/home/gitlab-runner/builds/7ab87121/0/questa-tensor/software/sparten/build'
-- Configuring incomplete, errors occurred!
See also "/home/gitlab-runner/builds/7ab87121/0/questa-tensor/software/sparten/build/CMakeFiles/CMakeOutput.log".
See also "/home/gitlab-runner/builds/7ab87121/0/questa-tensor/software/sparten/build/CMakeFiles/CMakeError.log".
ERROR: Job failed: exit status 1
```

**Comment by masterleinad** [comment]
This will be fixed in #3135.

**Comment by ndellingwood** [comment]
@dmdunla should be fixed on develop branch and will be part of the 3.2 release
